fix(developer): remove redundant check in LdmlKeyboardCompiler.validate() 🙀

srl295 opened the pull request:
- simplify LdmlKeyboardCompiler.validate() by removing a redundant check

@keymanapp-test-bot skip

Suggested by https://github.com/keymanapp/keyman/pull/11810#discussion_r1650176743

jahorton commented on developer/src/kmc-ldml/src/compiler/compiler.ts:
```diff
@@ -291,10 +291,14 @@ export class LdmlKeyboardCompiler implements KeymanCompiler {
    */
   private async lint(source: LDMLKeyboardXMLSourceFile, kmx: KMXPlus.KMXPlusFile) : Promise<boolean> {
+    if (!kmx || !source) {
+      return false;
```
So... if there's no source or no kmx, that's a linter failure?

rc-swag replied:
Well, I see that it isn't a linter failure just a broken pre-condition so bail returning false.

mcdurdin commented:
> Fixes: #11054

I don't understand how this PR fixes that?

Could you update the title to clarify what is being done here?

rc-swag reviewed (approved):
lgtm

srl295 commented:
> > Fixes: #11054
> 
> I don't understand how this PR fixes that?
> 
> Could you update the title to clarify what is being done here?

Updated.

I removed the "Fixes" - the feature work was in https://github.com/keymanapp/keyman/pull/11810 this is just an improvement to the fix.

keyman-server commented:
Changes in this pull request will be available for download in [Keyman version 18.0.92-alpha](https://keyman.com/downloads/releases/alpha/18.0.92)
